pkg/3scale/amp/component/system_monitoring.go

```diff
@@ -173,3 +165,26 @@ func SystemSidekiqPrometheusRules(ns string) *monitoringv1.PrometheusRule {
 		},
 	}
 }
+
+func (system *System) monitoringLabels() map[string]string {
+	labels := make(map[string]string)
+
+	for key, value := range system.Options.CommonLabels {
+		labels[key] = value
+	}
+
+	labels["monitoring-key"] = common.MonitoringKey
+	return labels
+}
+
+func (system *System) prometheusRulesMonitoringLabels() map[string]string {
+	labels := make(map[string]string)
+
+	for key, value := range system.Options.CommonLabels {
+		labels[key] = value
+	}
+
+	labels["prometheus"] = "application-monitoring"
```

Comment:
What's the purpose of this additional labels? Depending on it I'm wondering whether namespacing the label would be more appropriate.

I know the "monitoring-key" cannot be namespaced as we don't have control over it but if the `prometheus` and `role` labels are labels what we are arbitrarily defining here and are just informational maybe we could namespace them.

What do you think?

Reply:
Those are labels already existing in the first implementation of the prometheus rule objects. TBH, I cannot tell the reason to be there or who is using it. But I would not change it, unless there is a good reason to do so.

Reply:
:+1: makes sense. We can take a look at it in the future if we want to improve this or figure it out